src/main/scala/chisel3/simulator/ChiselSettings.scala

```diff
@@ -11,15 +72,92 @@ package chisel3.simulator
   * @param layerControl determines which [[chisel3.layer.Layer]]s should be
   * enabled _during Verilog elaboration_.
   */
-final class ChiselSettings(
+final class ChiselSettings[A <: RawModule](
```

Comment:
Consider privatizing the constructor and instead using a public factory method. Not a huge deal but IMO better style and makes some possible future refactors easier.

Reply:
Good idea. Changed to this in: 366de579b